gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/layouting/MtxDataSource.java

```diff
@@ -0,0 +1,195 @@
+import org.gradoop.common.model.impl.pojo.EPGMEdge;
+import org.gradoop.common.model.impl.pojo.EPGMVertex;
+import org.gradoop.flink.io.api.DataSource;
+import org.gradoop.flink.model.impl.epgm.GraphCollection;
+import org.gradoop.flink.model.impl.epgm.LogicalGraph;
+import org.gradoop.flink.model.impl.functions.epgm.Id;
+import org.gradoop.flink.util.GradoopFlinkConfig;
+
+import java.io.IOException;
+
+/**
+ * An importer for the
+ * <a href="https://math.nist.gov/MatrixMarket/formats.html">matrix-marked format</a> (.mtx)
+ */
+public class MtxDataSource implements DataSource {
+
+  /**
+   * Path to the input-file
+   */
+  private String path;
+  /**
+   * Gradoop config
+   */
+  private GradoopFlinkConfig cfg;
+  /**
+   * if true, skip the pre-processing (remove multi-edges, self-edges etc.)
+   */
+  private boolean skipPreprocessing;
+
+  /**
+   * Create new MTX-Datasource
+   *
+   * @param path Pathh to the input-file
+   * @param cfg  Gradoop-config to use
+   */
+  public MtxDataSource(String path, GradoopFlinkConfig cfg) {
+    this(path, cfg, false);
+  }
+
+  /**
+   * Create new MTX-Datasource
+   *
+   * @param path              Pathh to the input-file
+   * @param cfg               Gradoop-config to use
+   * @param skipPreprocessing if true, skip the pre-processing (remove multi-edges, self-edges etc.)
+   */
+  public MtxDataSource(String path, GradoopFlinkConfig cfg, boolean skipPreprocessing) {
+    if (path == null || cfg == null) {
+      throw new IllegalArgumentException("Stguments can not be null");
+    }
+    this.path = path;
+    this.cfg = cfg;
+    this.skipPreprocessing = skipPreprocessing;
+  }
+
+  @Override
+  public LogicalGraph getLogicalGraph() {
+    DataSet<EPGMVertex> vertices = cfg.getExecutionEnvironment().readTextFile(path)
+      .flatMap(new MtxVertexToVertex(cfg.getLogicalGraphFactory().getVertexFactory()))
+      .distinct(new Id<>());
+
+    DataSet<EPGMEdge> edges = cfg.getExecutionEnvironment().readTextFile(path)
+      .flatMap(new MtxEdgeToEdge(cfg.getLogicalGraphFactory().getEdgeFactory()));
+
+    if (!skipPreprocessing) {
+      edges = edges.filter((e) -> !e.getSourceId().equals(e.getTargetId()));
+      edges = edges.map(e -> {
+        if (e.getSourceId().compareTo(e.getTargetId()) > 0) {
+          GradoopId old = e.getSourceId();
+          e.setSourceId(e.getTargetId());
+          e.setTargetId(old);
+        }
+        return e;
+      }).distinct("sourceId", "targetId");
+    }
+
+    return cfg.getLogicalGraphFactory().fromDataSets(vertices, edges);
+  }
+
+  @Override
+  public GraphCollection getGraphCollection() throws IOException {
+    throw new UnsupportedOperationException("MTX does not support graph collections");
+  }
+
+  /**
+   * Returns true if the given line is a comment
+   *
+   * @param line The line to check
+   * @return true if comment
+   */
+  private static boolean isComment(String line) {
+    return line.startsWith("#") || line.startsWith("%");
```

Comment:
I think the matrix market format specification (https://math.nist.gov/MatrixMarket/formats.html) does not define the `#` symbol as a marker for a doc string, so `line.startsWith("#")` should be removed